Source/WebCore/style/RuleSet.cpp

```diff
@@ -473,6 +473,9 @@ Vector<Ref<const StyleRuleScope>> RuleSet::scopeRulesFor(const RuleData& ruleDat
         identifier = query.parent;
     };
 
+    // Order scopes from outermost to innermost.
+    queries.reverse();
```

Comment:
Can we append in reverse order directly? or is that not trivial?

Reply:
We can preprend into the `WTF::Vector` (`insert(0,value)`) Not sure if it's better though

Reply:
You can use a `Deque` to prepend efficiently.